pkg/client/options_test.go

```diff
@@ -292,3 +305,45 @@ var _ = Describe("ForceOwnership", func() {
 		Expect(*o.Force).To(Equal(true))
 	})
 })
+
+var _ = Describe("HasLabels", func() {
+	const invalidLongKey = "axahm2EJ8Phiephe2eixohbee9eGeiyees1thuozi1xoh0GiuH3diewi8iem7Nui"
+
+	It("Should produce hasLabels in given order", func() {
+		listOpts := &client.ListOptions{}
+
+		hasLabels := client.HasLabels([]string{"labelApe", "labelFox"})
+		hasLabels.ApplyToList(listOpts)
+		Expect(listOpts.LabelSelector.String()).To(Equal("labelApe,labelFox"))
+	})
+
+	It("Should produce an empty selector when given invalid input", func() {
```

Comment:
This is a bug IMHO. Given invalid input we should produce an invalid selector so this errors down the line rather than just silently dropping the selector.

Its fine not to fix that bug in here but please do not add a testcase for the presence of a bug :)

Reply:
I will remove the case as a quick fix.  
(It did puzzled for a while where the behavior of `HasLable.ApplyToList()` differs from that of `MatchingLabels`. Both of them are expected to produce an invalid selector on errors)